src/Core/ChangeInformation.vala

```diff
@@ -161,43 +88,22 @@ public class AppCenterCore.ChangeInformation : Object {
      }
 
     public void reset_progress () {
-        status = Pk.Status.SETUP;
+        status = Status.UNKNOWN;
+        status_description = _("Starting");
         progress = 0.0f;
-        last_progress = 0;
-        current_progress = 0;
-        current_status = 0;
-        progress_denom = 200.0f;
     }
 
-    public void ProgressCallback (Pk.Progress progress, Pk.ProgressType type) {
-        switch (type) {
-            case Pk.ProgressType.ALLOW_CANCEL:
-                can_cancel = progress.allow_cancel;
-                status_changed ();
-                break;
-            case Pk.ProgressType.ITEM_PROGRESS:
-                if (current_status == Pk.Status.SETUP) {
-                    current_status = (Pk.Status) progress.status;
-                    /* skipping package download, we have cached packages */
-                    if (current_status != Pk.Status.DOWNLOAD) {
-                        progress_denom = 100.0f;
-                    }
-                }
-                /* transaction changed so progress count is starting over */
-                else if ((Pk.Status) progress.status != current_status) {
-                    current_status = (Pk.Status) progress.status;
-                    current_progress = last_progress;
-                }
+    public void Callback (bool can_cancel, string status_description, uint progress, Status status) {
+        if (this.can_cancel != can_cancel || this.status_description != status_description || this.status != status) {
+            this.can_cancel = can_cancel;
+            this.status_description = status_description;
+            this.status = status;
+            status_changed ();
+        }
 
-                last_progress = progress.percentage;
-                double progress_sum = current_progress + last_progress;
-                this.progress = progress_sum / progress_denom;
-                progress_changed ();
-                break;
-            case Pk.ProgressType.STATUS:
-                status = (Pk.Status) progress.status;
-                status_changed ();
-                break;
+        if (this.progress != (progress / 100.0f)) {
```

Comment:
Is this needed to normalize this progress here instead of expecting from the backend to normalize it on it's own and then pass the normalized version to the method?